File: app_sample_java/src/main/java/com/joelzhu/lib/scanner/java/commands/CommandSleep.java
package com.joelzhu.lib.scanner.java.commands;

import com.joelzhu.lib.scanner.annotation.CompileScan;
import com.joelzhu.lib.scanner.java.ICommand;
import com.joelzhu.lib.scanner.java.utils.Constants;

/**
 * Command: human to sleep.
 *
 * @author JoelZhu
 * @since 2023-01-27
 */
@CompileScan(priority = 10)
public final class CommandSleep implements ICommand {
    @Override
    public String execute() {
        return "Sleeping... By default, priority = 10";
    }
}

File: app_sample_kotlin/src/main/java/com/joelzhu/lib/scanner/kotlin/commands/CommandSleep.kt
package com.joelzhu.lib.scanner.kotlin.commands

import com.joelzhu.lib.scanner.annotation.CompileScan
import com.joelzhu.lib.scanner.kotlin.ICommand

/**
 * Command: human to sleep.
 *
 * @author JoelZhu
 * @since 2023-03-13
 */
@CompileScan(priority = 4)
class CommandSleep : ICommand {
    override fun execute(): String {
        return "Sleeping... Priority = 4"
    }
}